slam-libraries/viam-cartographer/src/slam_service/slam_service_test.cc

```diff
@@ -0,0 +1,132 @@
+                   tolerance);
+        BOOST_TEST(overlapping_submaps_trimmer_2d.min_added_submaps_count() ==
+                   slamService.min_added_submaps_count);
+    } else {
+        BOOST_TEST(overlapping_submaps_trimmer_2d.fresh_submaps_count() == 0);
+        BOOST_TEST(overlapping_submaps_trimmer_2d.min_covered_area() == 0,
+                   tolerance);
+        BOOST_TEST(overlapping_submaps_trimmer_2d.min_added_submaps_count() ==
+                   0);
+    }
+    BOOST_TEST(ceres_scan_matcher_options.occupied_space_weight() ==
+                   slamService.occupied_space_weight,
+               tolerance);
+    BOOST_TEST(ceres_scan_matcher_options.translation_weight() ==
+                   slamService.translation_weight,
+               tolerance);
+    BOOST_TEST(ceres_scan_matcher_options.rotation_weight() ==
+                   slamService.rotation_weight,
+               tolerance);
+}
+
+BOOST_AUTO_TEST_CASE(OverwriteMapBuilderParameters_set_values_mapping) {
+    SLAMServiceImpl slamService;
+    slamService.map_rate_sec = std::chrono::seconds(60);
+    slamService.optimize_every_n_nodes = 9999;
+    slamService.num_range_data = 9999;
+    slamService.missing_data_ray_length = 9999.99;
+    slamService.max_range = 9999.99;
+    slamService.min_range = 9999.99;
+    slamService.max_submaps_to_keep = 9999;
+    slamService.fresh_submaps_count = 9999;
+    slamService.min_covered_area = 9999.99;
+    slamService.min_added_submaps_count = 9999;
+    slamService.occupied_space_weight = 9999.99;
+    slamService.translation_weight = 9999.99;
+    slamService.rotation_weight = 9999.99;
+    slamService.OverwriteMapBuilderParameters();
+    checkCartoMapBuilderParameters(slamService);
+}
+
+BOOST_AUTO_TEST_CASE(
+    OverwriteMapBuilderParameters_check_default_values_mapping) {
+    SLAMServiceImpl slamService;
+    slamService.map_rate_sec = std::chrono::seconds(60);
+    slamService.OverwriteMapBuilderParameters();
+    checkCartoMapBuilderParameters(slamService);
+}
+
+BOOST_AUTO_TEST_CASE(GetActionMode_mapping) {
+    SLAMServiceImpl slamService;
+    slamService.map_rate_sec = std::chrono::seconds(60);
+    BOOST_TEST((slamService.GetActionMode() == SLAMServiceActionMode::MAPPING));
+}
+
+/*
+BOOST_AUTO_TEST_CASE(GetActionMode_updating) {
+    // TODO: Add a test case GetActionMode for updating.Requires that an apriori
+    // map is detected and loaded. Will be implemented in this ticket:
+    // https://viam.atlassian.net/browse/DATA-114
+    BOOST_TEST((slamService.GetActionMode() ==
+SLAMServiceActionMode::UPDATING));
+}
+*/
+
+/*
+BOOST_AUTO_TEST_CASE(GetActionMode_localizing) {
+    // TODO: Once this scope doc is approved:
+    //
+https://docs.google.com/document/d/1RsT-c0QOtkMKa-rwUGY0-emUmKIRSaO3mzT1v6MtFjk/edit#
+    SLAMServiceImpl slamService;
+    slamService.map_rate_sec = -1; // Would need refactoring
+    BOOST_TEST((slamService.GetActionMode() ==
+SLAMServiceActionMode::LOCALIZING));
+}
+*/
+
```

Comment:
Added these so we don't forget to implement them once we've implemented aprior map loading or decided on how to implement localizing mode (once the current scope doc is approved).